src/coreclr/vm/siginfo.hpp

```diff
@@ -1208,6 +1184,51 @@ class MetaSig
         BYTE            m_CallConv;
 };  // class MetaSig
 
+// Attempts to parse the provided calling convention names to construct a
+// calling convention with known modifiers.
+//
+// The Add* functions return false if the type is known but cannot be applied.
+// Otherwise, they return true.
+class CallConvBuilder final
+{
+public:
+    enum CallConvModifiers
+    {
+        CALL_CONV_MOD_NONE = 0,
+        CALL_CONV_MOD_SUPPRESSGCTRANSITION = 0x1,
+        CALL_CONV_MOD_MEMBERFUNCTION = 0x2
+    };
+
+    struct State
+    {
+        CorInfoCallConvExtension CallConvBase;
+        CallConvModifiers CallConvModifiers;
+    };
```

Comment:
The `State` struct should be able to be private to `CallConvBuilder`.

Reply:
Yes I could. but then the internal templated `ProcessName` function would need to be a friend function. Making the nested type public seemed like a reasonable compromise between that and making it a high level one.

Reply:
I agree, that's a good reason. We can merge this as is then.